Description: qemu: Allow empty script path to <interface/>

commit 1d9ab0f04af310e52f80b4281751655bb3bb7601
Author: Michal Privoznik <mprivozn@redhat.com>
Date:   Thu Feb 2 14:16:20 2017 +0100

    qemu: Allow empty script path to <interface/>

    Before 9c17d665fdc5f (v1.3.2 - I know, right?) it was possible to
    have the following interface configuration:

      <interface type='ethernet'/>
        <script path=''/>
      </interface>

    This resulted in -netdev tap,script=,.. Fortunately, qemu helped
    us to get away with this as it just ignored the empty script
    path. However, after the commit mentioned above it's libvirtd
    who is executing the script. Unfortunately without special
    case-ing empty script path.

    Signed-off-by: Michal Privoznik <mprivozn@redhat.com>

Note: can be dropped >=3.1

Original-Author: Michal Privoznik <mprivozn@redhat.com>
Author: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/1665698
Forwarded: yes (backport from upstream 1d9ab0f0)
--- a/src/util/virnetdev.c
+++ b/src/util/virnetdev.c
@@ -2615,6 +2615,10 @@
     virCommandPtr cmd;
     int ret;
 
+    /* Not a bug! Previously we did accept script="" as a NO-OP. */
+    if (STREQ(script, ""))
+        return 0;
+
     cmd = virCommandNew(script);
     virCommandAddArgFormat(cmd, "%s", ifname);
     virCommandClearCaps(cmd);
